src/App.js

```diff
@@ -1,16 +1,81 @@
 import React from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
+import { useState } from 'react';
+import ColorChoices from './components/ColorChoice';
 
 const App = () => {
+  const [chatData, setChatData] = useState(chatMessages);
+
+  const updatedChatData = (updatedChatEntry) => {
+    const entries = chatData.map((entry) => {
+      if (entry.id === updatedChatEntry.id) {
+        return updatedChatEntry;
+      } else {
+        return entry;
+      }
+    });
+    setChatData(entries);
+  };
+
+  const numberOfLikes = chatData.filter((entry) => entry.liked === true).length;
+  const heartText = numberOfLikes < 2 ? '❤️' : '❤️s';
+
+  const updateChatColor = (updatedChatEnties) => {
+    setChatData(updatedChatEnties);
+  };
+
+  const defineColor = (data, mySender) => {
+    return data.filter((entry) => entry.sender === mySender)[0].color;
+  };
+
   return (
     <div id="App">
       <header>
-        <h1>Application title</h1>
+        <h1>
+          Chat between{' '}
+          <span className={defineColor(chatData, 'Vladimir')}>Vladimir</span>{' '}
+          and{' '}
+          <span className={defineColor(chatData, 'Estragon')}>Estragon</span>
```

Comment:
Tip: This is a little brittle since it means we can't re-use these components for a chat between different people.